src/components/ComboBox/ComboBox.test.tsx

```diff
@@ -0,0 +1,135 @@
+import ComboBox, { ComboBoxProps } from './ComboBox';
+
+const options = [{ value: '1999' }, { value: '2000' }, { value: '2001' }, { value: '2002' }];
+
+describe('ComboBox', () => {
+  runCommonTests(ComboBox, 'ComboBox');
+  const reqProps = {
+    labelText: 'My Test Label',
+    id: 'test-id',
+    options,
+    setInputValue: () => vitest.fn(),
+    inputValue: '',
+  };
+
+  it('will render a label value if passed', () => {
+    render(<ComboBox {...reqProps} />);
+    const label = screen.getAllByTestId('test-id-label');
+    expect(label[0]).toBeInTheDocument();
+  });
+
+  it('should render the dropdown list after clicking the trigger', async () => {
+    render(<ComboBox {...reqProps} />);
+    const trigger = screen.getAllByTestId('test-id-dropdown')[0];
+
+    expect(trigger).toBeInTheDocument();
+    await userEvent.click(trigger);
+
+    expect(screen.getByText('1999')).toBeInTheDocument();
+    expect(screen.getByText('2000')).toBeInTheDocument();
+    expect(screen.getByText('2001')).toBeInTheDocument();
+    expect(screen.getByText('2002')).toBeInTheDocument();
+    expect(screen.queryByText('No Options')).not.toBeInTheDocument();
+  });
+
+  it('should be able to select an option from the dropdown by clicking on it', async () => {
+    const mockSetInputValue = vi.fn();
+    render(<ComboBox {...{ ...reqProps, setInputValue: mockSetInputValue }} />);
+    const trigger = screen.getAllByTestId('test-id-dropdown')[0];
+
+    expect(trigger).toBeInTheDocument();
+    await userEvent.click(trigger);
+
+    const option = screen.getByText('2000');
+    expect(option).toBeInTheDocument();
+    await userEvent.click(option);
+
+    expect(mockSetInputValue).toHaveBeenCalledWith('2000');
+  });
+
+  it('should be able to select an option from the dropdown by keyboard navigation', async () => {
+    const mockSetInputValue = vi.fn();
+    render(<ComboBox {...{ ...reqProps, setInputValue: mockSetInputValue }} />);
+    const trigger = screen.getAllByTestId('test-id-dropdown')[0];
+
+    await userEvent.click(trigger);
+
+    await userEvent.keyboard('{ArrowDown}');
+    await userEvent.keyboard('{ArrowDown}');
+    await userEvent.keyboard('{ArrowDown}');
+    await userEvent.keyboard('{Enter}');
+
+    expect(mockSetInputValue).toHaveBeenCalledWith('2002');
+  });
+
+  it('should display No Options when input does not match any options', async () => {
+    const mockSetInputValue = vi.fn();
+    render(<ComboBox {...{ ...reqProps, setInputValue: mockSetInputValue, inputValue: 'non-existent' }} />);
+    const trigger = screen.getAllByTestId('test-id-dropdown')[0];
+
+    await userEvent.click(trigger);
+    const input = screen.getAllByTestId('test-id-input')[0];
+
+    await userEvent.click(input);
+    await userEvent.type(input, 'HI');
+
+    expect(screen.queryByText('No Options.')).toBeInTheDocument();
+  });
+
+  it('should clear input value when the close button is clicked', async () => {
```

Comment:
This has a similar test name to the test below and this test does not click on the close button
```suggestion
  it('should clear input value when user clicks outside of input', async () => {
```

It is debatable if this is desirable UX. What if the user wanted to copy text from somewhere else on the page and include? Can ask the team what they think. Not blocking approval over it.